include/hip/hcc_detail/hip_runtime_api.h

```diff
@@ -1480,9 +1486,16 @@ inline
 __attribute__((visibility("hidden")))
 hipError_t hipGetSymbolAddress(void** devPtr, const void* symbolName) {
     //HIP_INIT_API(hipGetSymbolAddress, devPtr, symbolName);
+#ifdef __cplusplus
     hip_impl::hip_init();
     size_t size = 0;
     return hip_impl::read_agent_global_from_process(devPtr, &size, (const char*)symbolName);
+#else
+    hip_init();
+    // read_agent_global_from_process is defined in cpp file hence returning error here to indicate
+    // func not implemented for 'c' code
+    return hipErrorTbd;   
```

Comment:
Let us define and use proper error code for this.

Reply:
Agree, will take it in follow-up PR as I have to make sure that it is not causing any regression

Reply:
It would be preferred to just compile this function out in C then.  It won't cause any "regression" if no new C test is being added to exercise this function